### ISSUE TITLE
Unbiased Learning-to-Rank from biased feedback data

### ISSUE THREAD

**Issue by eggie5** [tensorflow/ranking#9]
The README references the work from Joachims in "Unbiased Learning-to-Rank from biased feedback data." I was expecting to see some type of IPS implementation in the code, but I can't find any reference to the work. What am I missing?

**Comment by xuanhuiwang** [comment]
Thanks for the question. Support of unbiased LTR is a less-visible feature in tf-ranking.

In general, LTR does not take per-item weights, especially for pairwise loss functions. For unbiased LTR, the loss is designed asymmetrically in tf-ranking. Please see the comment and code here: https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/python/losses.py#L518 and the paper https://arxiv.org/pdf/1608.04468.pdf and Section 3.2 at paper https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/46485.pdf.

How to estimate the propensity is not included in this library as it may need swapping results. Once the propensity is estimated, Inverse Propensity Scores can be passed to the "weights" argument in the loss function as a per-example or per-list tensor.

**Comment by Ekkalak-T** [comment]
Thanks for the question and answer. I am trying to find the "weights" argument in loss function.
However, I found three weights in "make_loss_fn" function. Which one is correct?
(loss_weights, weights_feature_name, lambda_weight, )

https://github.com/tensorflow/ranking/blob/099b17b11e7f699f30bcdea18a4323df7827fde8/tensorflow_ranking/python/losses.py#L52

**Comment by ramakumar1729** [comment]
@Ekkalak-T : The `weights_feature_name` argument specifies the weight for inverse propensity weights. The weight is assumed to be one of the features in the features dict generated by the input reader.
